lib/spack/spack/audit.py

```diff
@@ -203,6 +204,40 @@ def _search_duplicate_compilers(error_cls):
     return errors
 
 
+def compiler_def_to_string(c):
+    """Return a string composed of the paths, os, and target"""
+    compiler = c.get("compiler", {})
+    paths = compiler.get("paths", {})
+    return "_".join(
+        [
+            paths.get("cc", "None") or "None",
+            paths.get("cxx", "None") or "None",
+            paths.get("f77", "None") or "None",
+            paths.get("fc", "None") or "None",
+            compiler.get("operating_system", "None"),
+            compiler.get("target", "None"),
```

Comment:
Actually, I'm wondering if this should include every configuration detail about the compiler, rather than only paths, os, and arch.  @kwryankrattiger Mentioned he sometimes copies a compiler config and changes the name, only to change what flags are passed, which seems valid to me.  But what I have currently would cause that to fail the audit.